ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilterInputField.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+import { SearchInput, SelectOption } from '@patternfly/react-core';
+import { SelectedEntity } from './EntitySelector';
+import { SelectedAttribute } from './AttributeSelector';
+import {
+    CompoundSearchFilterConfig,
+    SearchFilterAttribute,
+    SelectSearchFilterAttribute,
+} from '../types';
+
+import CheckboxSelect from './CheckboxSelect';
+
+export type InputFieldValue = string | number | undefined | string[];
+export type InputFieldOnChange = (value: InputFieldValue) => void;
+
+export type CompoundSearchFilterInputFieldProps = {
+    selectedEntity: SelectedEntity;
+    selectedAttribute: SelectedAttribute;
+    value: InputFieldValue;
+    onSearch: InputFieldOnChange;
+    onChange: InputFieldOnChange;
+    config: Partial<CompoundSearchFilterConfig>;
+};
+
+function isSelectType(
+    attributeObject: SearchFilterAttribute
+): attributeObject is SelectSearchFilterAttribute {
+    return attributeObject.inputType === 'select';
+}
+
+function CompoundSearchFilterInputField({
+    selectedEntity,
+    selectedAttribute,
+    value,
+    onSearch,
+    onChange,
+    config,
+}: CompoundSearchFilterInputFieldProps) {
+    if (!selectedEntity || !selectedAttribute) {
+        return null;
+    }
+
+    const attributeObject: SearchFilterAttribute =
+        config[selectedEntity]?.attributes[selectedAttribute];
+
+    if (!attributeObject) {
+        return null;
+    }
```

Comment:
I think 1. Render nothing is appropriate in this case. In some situations I think it makes sense to define the props in a way that we _can't_ pass an invalid value, but given that the type needs to be `Partial` I don't think this is easily possible.

Since this would mostly likely be a developer error, there isn't anything a user could do with a message that would fix the problem. We should be able to catch the nothing case here with tests though on the pages this component is used.